daemon/src/file_asset_source.rs

```diff
@@ -1207,140 +1209,150 @@ impl FileAssetSource {
-                    let metadata = if let Some(mut import_output) = import_output {
-                        // TODO store reported errors and warnings in metadata
-                        if let Some(import_op) = import_output.import_op {
-                            for error in &import_op.errors {
-                                log::error!("Import errors {:?}: {:?}", p.source, error);
-                            }
-                            for warning in &import_op.warnings {
-                                log::warn!("Import warning {:?}: {:?}", p.source, warning);
-                            }
-                        }
-                        // put import artifact in cache if it doesn't have unresolved refs
-                        if !import_output.assets.is_empty() {
-                            let mut txn = self
-                                .artifact_cache
-                                .rw_txn()
-                                .await
-                                .expect("failed to get cache txn");
-
-                            for asset in import_output.assets.iter_mut() {
-                                if asset.is_fully_resolved() {
-                                    if let Some(serialized_asset) = asset.serialized_asset.as_mut()
-                                    {
-                                        serialized_asset.metadata.id =
-                                            ArtifactId(utils::calc_import_artifact_hash(
-                                                &asset.metadata.id,
-                                                import.import_hash().unwrap(),
-                                                serialized_asset
-                                                    .metadata
-                                                    .load_deps
-                                                    .iter()
-                                                    .chain(
-                                                        serialized_asset.metadata.build_deps.iter(),
-                                                    )
-                                                    .map(|dep| dep.expect_uuid()),
-                                            ));
-                                        log::trace!(
-                                            "caching asset {:?} from file {:?} with hash {:?}",
-                                            asset.metadata.id,
-                                            p.source,
-                                            serialized_asset.metadata.id
-                                        );
-                                        self.artifact_cache.insert(&mut txn, serialized_asset);
-                                    } else {
-                                        log::trace!("asset {:?} from file {:?} did not return serialized asset: cannot cache", asset.metadata.id, p.source );
+        self.runtime.scope(|scope| {
```

Comment:
This whole block is questionable but it seems to work. I was trying to replace `Scope` with a task pool.

Reply:
Yeah I think it looks all right 👍